changelog/5515.feature.rst

```diff
@@ -0,0 +1,15 @@
+Allow selective auto-indentation of multiline log messages.
+
+Supports command line option ``--log-auto-indent [value]``
+and config option ``log_auto_indent`` to set the default for all logging.
+
+Supports passing kwarg ``extra={"auto_indent": [value]}`` to
+calls to ``log.log()`` to specify auto-indentation behavior for
+a specific entry in the log.
+
+``[value]`` can be:
+    * True or On - Dynamically auto-indent multiline log messages
+    * False or Off or 0 - Do not auto-indent multiline log messages
+    * [integer] - auto-indent multiline log messages by [integer] spaces
```

Comment:
Great that you've added support for configuring the the indentation level explicitly!

Reply:
Thanks

Reply:
I'd rather see a link from the changelog to docs, instead of having it expanded/explained that much there.
(people apparently (according to #5952 at least) will find it then in the changelog, and might see an out-of-date documenation)

Reply:
> s/integer/positive integer/ ?

Agreed

Reply:
> I'd rather see a link from the changelog to docs, instead of having it expanded/explained that much there.

Can you provide more detail what you'd like to see here, or perhaps the exact text you prefer? I'm not sure what you're looking for.

Reply:
@earthman1
Not sure what's unclear.. the changelog basically documents the feature/change, while that should be in the docs themselves instead (where it gets updated in the future).

I.e. https://github.com/pytest-dev/pytest/blob/817a42d48fb00cf50553facecdbbc24dadabf36c/changelog/5515.feature.rst should go into https://github.com/pytest-dev/pytest/blame/914a9465aba501e776282961fb3317837772435d/doc/en/reference.rst#L1196, and then the changelog should link to that (IIRC that's not possible though due to pre-commit failing on it then.. https://github.com/pytest-dev/pytest/issues/4283).  So assuming #4283 is still a problem, it should just mention/describe the setting briefly, but not document it there.